lib/client/player_test.go

```diff
@@ -133,7 +133,102 @@ func TestPlayPause(t *testing.T) {
 		t.Fatal("timed out waiting for player to complete")
 	case <-p.stopC:
 	}
-	require.True(t, p.Stopped())
+	require.True(t, p.Stopped(), "p.Stopped() returned false when it should have returned true")
+}
+
+func TestEndPlaybackWhilePlaying(t *testing.T) {
+	c := clockwork.NewFakeClock()
+
+	// in this test, we let the player play 2 of the 3 events,
+	// then pause it and verify the pause state before resuming
+	// playback for the final event.
+	events := printEvents(100, 200, 300)
+	var stream []byte // intentionally empty, we dont care about stream contents here
+	p := newSessionPlayer(events, stream, testTerm(t))
+	p.clock = c
+
+	p.Play()
+
+	// wait for player to see the first event and apply the delay
+	c.BlockUntil(1)
+
+	// advance the clock:
+	// at this point, the player will write the first event
+	c.Advance(100 * time.Millisecond)
+
+	// wait for the player to sleep on the 2nd event
+	c.BlockUntil(1)
+
+	// end playback
+	p.EndPlayback()
+
+	// advance the clock again:
+	// the player will write the second event and
+	// then realize that it's been asked to end playback
+	c.Advance(100 * time.Millisecond)
+
+	// check that stopC was written to
+	select {
+	case <-time.After(2 * time.Second):
+		t.Fatal("timed out waiting for player to complete")
+	case <-p.stopC:
+		require.True(t, p.Stopped(), "p.Stopped() returned false when it should have returned true")
+	}
+}
+
+// TestEndPlaybackWhilePaused tests that playback can be ended
+// by the user pressing ctrl+C or ctrl+D while the playback is
+// paused.
```

Comment:
This is not exactly true. Nothing about test covers ctrl+C or ctrl+D behavior.

More accurately, this tests that playback can be ended by calling `EndPlayback` while playback is paused.